python/lsst/ts/m2gui/controltab/tab_actuator_control.py

```diff
@@ -33,7 +57,516 @@ class TabActuatorControl(TabDefault):
+            ),
+            "clear": set_button(
+                "Clear", self._callback_script_command, CommandScript.Clear
+            ),
+        }
+
+        self._target_displacement = self._create_target_displacement()
+        self._displacement_unit_selection = self._create_displacement_unit_selection()
+
+        self._ring_selection = self.create_combo_box_ring_selection()
+        self._buttons_actuator_selection = self._create_buttons_actuator_selection()
+        self._buttons_actuator_selection_support = {
+            "select_ring": set_button("Select Ring", self._callback_select_ring),
+            "clear_selection": set_button(
+                "Clear Selection", self._callback_clear_selection
+            ),
+        }
+
+        self._buttons_actuator = {
+            "start": set_button("Start", self._callback_actuator_start),
+            "stop": set_button(
+                "Stop", self._callback_actuator_command, CommandActuator.Stop
+            ),
+            "pause": set_button(
+                "Pause", self._callback_actuator_command, CommandActuator.Pause
+            ),
+            "resume": set_button(
+                "Resume", self._callback_actuator_command, CommandActuator.Resume
+            ),
+        }
+
+        self._set_widget_and_layout()
+
+        self._set_signal_script(self.model.signal_script)
+        self._set_signal_detailed_force(
+            self.model.utility_monitor.signal_detailed_force
+        )
+
+    def _create_progress_bar(self, value_min, value_max, value_init):
+        """Create the progress bar. This will show the execution progress of
+        script in the cell controller.
+
+        Parameters
+        ----------
+        value_min : `float` or `int`
+            Minimum value.
+        value_max : `float` or `int`
+            Maximum value.
+        value_init : `float` or `int`
+            Initial value.
+
+        Returns
+        -------
+        progress_bar : `PySide2.QtWidgets.QProgressBar`
+            Progress bar.
+        """
+
+        progress_bar = QProgressBar()
+        progress_bar.setRange(value_min, value_max)
+        progress_bar.setValue(value_init)
+
+        return progress_bar
+
+    @Slot()
+    def _callback_script_load_script(self, file_name=""):
+        """Callback of the load-script button in script control. The cell
+        controller will load the script.
+
+        Parameters
+        ----------
+        file_name : `str`, optional
+            File name. This is used for the unit test only. (the default is "")
+
+        Returns
+        -------
+        name : `str`
+            Script name.
+        """
+
+        if file_name == "":
+            file_name = QFileDialog.getOpenFileName(caption="Select script file")[0]
+
+        name = ""
+        if file_name != "":
+            # Only the file name is important
+            file_path = Path(file_name)
+            name = file_path.name
+
+            is_successful = run_command(
```

Comment:
When Harris delivered the code, the M2 EUI in server only needs to provide the file name to M2 cell in cRIO (and you need to select the file from the server). The cRIO needs to hold the script file in the `/u/script/` directory as a hard-coded path. This is weird but I do not have the intention to modify the M2 cell LabVIEW code at this moment (since it will be replaced by C++ code in the final).

There is no default scripts when Harris delivered the code. The user needs to base on the tool provided by Harris to create the script file to be used in the M2 cell. This is why I leave the choice to the user to decide which file he/she wants to select and there is no default directory or files to choose in the server side.

I think after the C++ code is done, the M2 EUI should transfer the details in JSON packet to M2 cell C++ code instead of holding a duplicated script in the cRIO. This will be done at that time, but not now.

Thanks!